manager/allocator/network.go

```diff
@@ -1071,6 +1072,16 @@ func (a *Allocator) procTasksNetwork(ctx context.Context, onRetry bool) {
 	}
 }
 
+// IsBuiltInNetworkDriver returns whether the passed driver is an internal network driver
+func (a *Allocator) IsBuiltInNetworkDriver(name string) bool {
+	return a.netCtx.nwkAllocator.IsBuiltInDriver(name)
+}
+
+// PredefinedNetworks returns the list of predefined network structures for a given network model
+func PredefinedNetworks() []networkallocator.PredefinedNetworkData {
```

Comment:
The [call in `becomeLeader`](https://github.com/docker/swarmkit/blob/master/manager/manager.go#L947) is before [the allocator is created](https://github.com/docker/swarmkit/blob/master/manager/manager.go#L975). I wasn't terribly comfortable with reordering to that extent, but I could give it a go if you think it seems worthwhile.

FYI in my CNI POC this ends up taking the NetworkModel as a parameter and `switch`ing over it, which I'm not incredibly happy with TBH.